contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/DrillHiveMetaStoreClient.java

```diff
@@ -209,8 +213,10 @@ private DrillHiveMetaStoreClient(final HiveConf hiveConf) throws MetaException {
       throws TException {
     try {
       return mClient.getAllTables(dbName);
+    } catch (MetaException | UnknownDBException e) {
+      throw e;
     } catch (TException e) {
-      logger.warn("Failure while attempting to get hive tables", e);
+      logger.warn("Failure while attempting to get hive tables. Retries once.", e);
```

Comment:
If it is Hive issue in reconnect(), can we close the client explicitly here until we upgrade the Hive version? That way we can avoid more leaks.